### ISSUE TITLE
notify_only does not work for sarra process

### ISSUE THREAD

**Issue by habilinour** [MetPX/sarracenia#321]
If you set notify_only to True in sarra config, it throws an error and the process exits

**Comment by habilinour** [comment]
I seems the the code was designed not to accept that option for sarra process
```
# always download ...

        if self.notify_only :
           self.logger.error("sarra notify_only True")
           os._exit(1)
```

**Comment by petersilva** [comment]
It´s odd, we just need to remove the if statement and run some tests to ensure it does not break anything.  I don´t know why it was there in the first place.  I think it is because Michel and I misunderstood one another.  I was saying that sarra should be used in certain circumstances, as a guideline, and he took it as a constraint, so implemented enforcement.

**Comment by benlapETS** [comment]
@petersilva your fix seems to do the right thing, I will add unit testing to insure notify_only (also no_download) option works as expected this time (no misunderstanding).

**Comment by benlapETS** [comment]
Ok from our specs ([Delivery-specifications](https://github.com/MetPX/sarracenia/blob/master/doc/sr_subscribe.1.rst#delivery-specifications)):

> no_download|notify_only \<boolean\> (default: off)

This is ambiguous. Do we accept boolean only with notify_only or with no_download too?

From our parser in sr_config:
>                 elif words0 in ['notify_only','n','no_download']: # See: sr_subscribe.1  
>                      self.logger.debug("option %s" % words[0])
>                      self.notify_only = True
>                      n = 1

We only treat it as one word option and where is "n" option in the specification ?

**Comment by benlapETS** [comment]
Should I align the spec to the code or the code to the spec ? Normally I wouldn't bother, the spec must be the reference and the code should adapt to it. But here i am unsure, this code is wrong and is there since 2016, maybe with some wrong usage and changing it might break something... but letting this like that seems just very poor coding practice...

**Comment by petersilva** [comment]
notify_only is much older than 2016... I think it goes all the way back to 2008... It is one of the original 3 for dd_subscribe.

There is a typo in the man page, missing square brackets. it should be:

```
  [ no_download | notify_only ] <boolean>   

```
The two words are synonyms and are correctly treated identically in the config parser.  
As with all such flag options having the option alone indicates True, but the value can optionally be explicitly stated.   

The short versions of the options are, by convention the first letter of each word. The short forms are not listed in the man page explicitly, for brevity.  For the letter 'n' ... the convention, throughout the parser, is "first come first serve" for short options.  'n' won ... should probably add ´nd´ and ´no´ to the list, as they make more sense than just ´n´.  Would have to look in the rest of the parser to see if there is a conflict in use of 'n'  I think analysts have been using -n forever, so would not take it away.  It is also listed when one types "sr_subscribe -h" near the top.

**Comment by benlapETS** [comment]
I agree with all you said, it's common sense. Also with the bracket it make more sense. Except for one thing: If somebody write [ no_download | notify_only ] False, it will turn it to True... which make no sense.

**Comment by benlapETS** [comment]
then what is the recommandation for the shorthand... we add -no and -nd alongside with -n and notify deprecation, recommand transition ?

**Comment by petersilva** [comment]
OK that is an error/bug.  look at any other boolean flag and there is standard handling.  Look at *accept_unmatched* ... it has the proper logic.  (invoking isTrue if an argument is present.)
for the abbreviations.  we can add the new ones, but should not get rid of old one... it is there
because it was the first one... it is OK as-is.

**Comment by benlapETS** [comment]
I already knew what to do, just wanted your confirmation...

... and for not getting rid of confusing option is a problem for me. Adding always new things without deprecating anything is a problem with Sarracenia. It is making the system more and more confusing over time. I suggest thinking of a strategy to get rid of confusing option that piled up with time.

**Comment by petersilva** [comment]
the analysts use that option a lot.  it is not confusing to them... there are other one letter options, such as 'a' for action, 'b' for broker, 'r' for randomize... I don't think the confusion is large enough to warrant breakage.  I think 'n' is good as-is, but the bad parsing is a genuine bug.

**Comment by benlapETS** [comment]
OK I understand your arguments.

**Comment by petersilva** [comment]
included in v2.20.04b1
